pkg/git/git.go

```diff
@@ -57,6 +59,23 @@ func NewCLIGit(cfg CLIGitConfig, r runner.Runner) *CLIGit {
 // Make sure CLIGit implements all the required methods.
 var _ Git = &CLIGit{}
 
+// Clone will take a repo location and clone it into a given folder.
+func (g *CLIGit) Clone(repo, branch, location string) error {
+	args := []string{
+		"clone",
+		"--branch",
+		branch,
+		"--depth",
+		"1",
+		repo,
+		location,
+	}
+	if err := g.runGitCmd(args...); err != nil {
+		return fmt.Errorf("failed to run clone: %w", err)
+	}
+	return nil
+}
+
```

Comment:
if clone fails, shall we attempt to cleanup the `location`? Since we don't know how far it got during failing

Reply:
ah your doing this outside the func already, nevermind!